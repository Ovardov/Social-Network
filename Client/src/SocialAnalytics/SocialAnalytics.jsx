import React from 'react';
import styles from './social-analystics.module.scss'

function SocialAnalytics({likes, comments}) {
    return (
        <section className={styles['social-container']}>
            <a href="#" className={styles.like}>
                <i className="far fa-heart"></i>
                {likes.length}
            </a>

            <a href="#" className={styles.comment}>
                <i className="far fa-comment"></i>
                {comments.length}
            </a>
        </section>
    )
}

export default SocialAnalytics;