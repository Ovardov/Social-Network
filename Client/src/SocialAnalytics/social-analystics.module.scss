.social-container {
    margin-left: auto;
    display: flex;

    & > div {
        display: flex;
        justify-content: flex-end;
        align-items: center;
        text-decoration: none;

        & i {
            margin-right: 0.3em;
        }

        &:not(:last-child) {
            margin-right: 0.7em;
        }

        &.like>i {
            color: var(--color-red-main);
        }

        &.comment>i {
            color: var(--color-blue-main);
        }
    }
}