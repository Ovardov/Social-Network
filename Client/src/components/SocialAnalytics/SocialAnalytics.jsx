import React from 'react';
import styles from './social-analystics.module.scss'

function SocialAnalytics({likes, comments}) {
    return (
        <section className={styles['social-container']}>
            <div className={styles.like}>
                <i className="far fa-heart"></i>
                {likes.length}
            </div>

            <div className={styles.comment}>
                <i className="far fa-comment"></i>
                {comments.length}
            </div>
        </section>
    )
}

export default SocialAnalytics;