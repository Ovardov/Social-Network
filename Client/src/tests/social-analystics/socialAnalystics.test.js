import React from 'react'
import renderer from 'react-test-renderer';
import SocialAnalytics from '../../components/SocialAnalytics/SocialAnalytics';

jest.mock("react-router-dom", () => ({
    Link: "Link"
}));

describe('SocialAnalytics component', () => {
    test('should render 0 likes and 0 comments', () => {
        const component = renderer.create(
            <SocialAnalytics likes={[]} comments={[]} />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('should render 5 likes and 0 comments', () => {
        const component = renderer.create(
            <SocialAnalytics likes={['first', 'second', 'third', 'fourth', 'fifth']} comments={[]} />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('should render 0 likes and 5 comments', () => {
        const component = renderer.create(
            <SocialAnalytics likes={[]} comments={['first', 'second', 'third', 'fourth', 'fifth']} />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });


    test('should render 5 likes and 5 comments', () => {
        const component = renderer.create(
            <SocialAnalytics
                likes={['first', 'second', 'third', 'fourth', 'fifth']}
                comments={['first', 'second', 'third', 'fourth', 'fifth']} />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
})
