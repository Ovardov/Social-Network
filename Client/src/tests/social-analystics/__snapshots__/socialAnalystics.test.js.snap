// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialAnalytics component should render 0 likes and 0 comments 1`] = `
<section
  className="social-container"
>
  <div
    className="like"
  >
    <i
      className="far fa-heart"
    />
    0
  </div>
  <div
    className="comment"
  >
    <i
      className="far fa-comment"
    />
    0
  </div>
</section>
`;

exports[`SocialAnalytics component should render 0 likes and 5 comments 1`] = `
<section
  className="social-container"
>
  <div
    className="like"
  >
    <i
      className="far fa-heart"
    />
    0
  </div>
  <div
    className="comment"
  >
    <i
      className="far fa-comment"
    />
    5
  </div>
</section>
`;

exports[`SocialAnalytics component should render 5 likes and 0 comments 1`] = `
<section
  className="social-container"
>
  <div
    className="like"
  >
    <i
      className="far fa-heart"
    />
    5
  </div>
  <div
    className="comment"
  >
    <i
      className="far fa-comment"
    />
    0
  </div>
</section>
`;

exports[`SocialAnalytics component should render 5 likes and 5 comments 1`] = `
<section
  className="social-container"
>
  <div
    className="like"
  >
    <i
      className="far fa-heart"
    />
    5
  </div>
  <div
    className="comment"
  >
    <i
      className="far fa-comment"
    />
    5
  </div>
</section>
`;
